lib/algos/heap/Heap.ts

```diff
@@ -0,0 +1,124 @@
+export enum HeapOrder {
+    Min = -1,
+    Max = 1,
+}
+
+export enum CompareResult {
+    LT = -1,
+    EQ = 0,
+    GT = 1,
+}
+
+export type CompareFunction = (x: any, y: any) => CompareResult;
+
+export class Heap {
```

Comment:
I am curious, what was the motivation to implement a heap in Arsenal? Don't we want to reuse existing heaps from proven open source projects?
Also, I am not sure (yet) to understand how it links to the addition of `NewerNoncurrentVersions`, is it to keep track of the objects?

Reply:
@williamlardier
not sure about introducing a new dependency for this;  the heap structure should be a relatively simple structure to not need to  do so.
yes. the heap will be used to keep track of the `latest` objects. In the case of `NewNoncurrentVersion`, a heap is used over a list/array due to the asynchronous methods needed to determine if a version needs to be expired.

Reply:
Thanks for the explanation!